src/flows/context/flow.query.tsx

```diff
@@ -374,20 +374,21 @@ export const FlowQueryProvider: FC = ({children}) => {
           return a
         }, {})
     )
+
     Promise.all(
       map
         .filter(q => !!q.visual)
         .map(stage => {
           return query(stage.visual, stage.scope)
             .then(response => {
-              setStatuses({[stage.id]: RemoteDataState.Done})
               setResult(stage.id, response)
+              setStatuses({[stage.id]: RemoteDataState.Done})
             })
             .catch(e => {
-              setStatuses({[stage.id]: RemoteDataState.Error})
               setResult(stage.id, {
                 error: e.message,
               })
+              setStatuses({[stage.id]: RemoteDataState.Error})
```

Comment:
not sure about the level of effort, but should these be coupled to reduce some of that enqueueing you mentioned? Like consolidate the status and result into 1 result object with 2 properties?